File: src/main/java/com/meepoffaith/hextrapats/casting/actions/sets/OpEmptyNumSet.java
package com.meepoffaith.hextrapats.casting.actions.sets;

import at.petrak.hexcasting.api.casting.eval.CastingEnvironment;
import at.petrak.hexcasting.api.casting.iota.Iota;
import com.meepoffaith.hextrapats.casting.bases.ConstMediaActionBase;
import com.meepoffaith.hextrapats.casting.bases.HexIotaStack;
import com.meepoffaith.hextrapats.casting.iota.DoubleSetIota;
import com.meepoffaith.hextrapats.casting.iota.DoubleSetIota.DoubleSet;

import java.util.HashSet;
import java.util.List;

public class OpEmptyNumSet extends ConstMediaActionBase{
    public int argc = 0;
    public long mediaCost = 0L;

    @Override
    public List<? extends Iota> execute(HexIotaStack stack, CastingEnvironment ctx){
        return asActionResult(new DoubleSetIota(new DoubleSet()));
    }
}

File: src/main/kotlin/com/meepoffaith/hextrapats/casting/actions/sets/OpEmptyNumSet.kt
package com.meepoffaith.hextrapats.casting.actions.sets

import at.petrak.hexcasting.api.casting.castables.ConstMediaAction
import at.petrak.hexcasting.api.casting.eval.CastingEnvironment
import at.petrak.hexcasting.api.casting.iota.Iota
import com.meepoffaith.hextrapats.casting.iota.DoubleSetIota

class OpEmptyNumSet : ConstMediaAction {
    override val argc = 0
    override fun execute(args: List<Iota>, env: CastingEnvironment): List<Iota> {
        return listOf(DoubleSetIota(DoubleSetIota.Companion.DoubleSet()))
    }
}
